File: src/main/java/com/extendedae_plus/network/base/SPacketGeneric.java
package com.extendedae_plus.network.base;

import net.minecraft.client.player.LocalPlayer;
import net.neoforged.api.distmarker.Dist;
import net.neoforged.api.distmarker.OnlyIn;
import net.neoforged.neoforge.network.handling.IPayloadContext;

public interface SPacketGeneric extends PacketGeneric {
    @OnlyIn(Dist.CLIENT)
    void handleClient(final LocalPlayer player);

    default void handle(final IPayloadContext context) {
        if (context.player() instanceof LocalPlayer player)
            this.handleClient(player);
    }
}

File: src/main/kotlin/com/extendedae_plus/network/base/SPacketGeneric.kt
package com.extendedae_plus.network.base

import net.minecraft.client.player.LocalPlayer
import net.neoforged.neoforge.network.handling.IPayloadContext

interface SPacketGeneric : PacketGeneric {
    fun handleClient(player: LocalPlayer)

    fun handle(context: IPayloadContext) {
        if (context.player() is LocalPlayer)
            this.handleClient(context.player() as LocalPlayer)
    }
}
